### PR TITLE
Startblock

### LINKED ISSUE #426: start
Start a containerVM and launch the main container process

Should start the main container process - either specified on command line or from image metadata (depends on #423)
Should publish the success/failure launching the container process
Should publish the return value of exited container process after successful launch
Docker start command should block on the cli until success/failure of launch is confirmed

### PR DESCRIPTION
**Pull request by hickeng**
Blocks until the container process has been launched. This will either return:
* nil: no error - process launched successfully
* err: process failed to launch - err contains the error message from the tether.

Fixes #426

### CODE REVIEW

File: portlayer/exec/container.go [hunk 1]
```diff
@@ -135,5 +139,36 @@ func (c *Container) Start(ctx context.Context) error {
...
+			}
+		}
+		return false
+	}
+
+	// Wait some before giving up...
+	ctx, cancel := context.WithTimeout(ctx, propertyCollectorTimeout)
+	defer cancel()
+
+	c.vm.WaitForExtraConfig(ctx, waitFunc)
```
Comment (dougm) [inline portlayer/exec/container.go]: Looks like you should still check err here, in case property.Wait() fails:
```go
if err != nil {
   return err
}
```

**Comment by fdawg4l** [comment]
Is it possible to add a test?  I don't think it would be trivial if at all but thought it worth asking.

**Comment by hickeng** [comment]
@fdawg4l not until @dougm has the sim up and running - it depends on property collector. I have added a test for the field that we block on into the tether tests, specifically in this case for the missing binary test.

**Comment by dougm** [comment]
lgtm

**Comment by fdawg4l** [comment]
lgtm

**Comment by hickeng** [comment]
LGTM

**Pull request merged**